csharp/Managed/Pd.cs

```diff
@@ -0,0 +1,202 @@
+		{
+			Inputs = inputChannels;
+			Outputs = outputChannels;
+			SampleRate = sampleRate;
+			//SetupHooks();
+			PInvoke.libpd_init ();
+			foreach (string path in searchPaths ?? Enumerable.Empty<string>()) {
+				PInvoke.add_to_search_path (path);
+			}
+		}
+
+		private void SetupHooks ()
+		{
+			//PrintHook = new LibPDPrintHook(RaisePrintEvent);
+			//PInvoke.set_printhook(PrintHook);
+
+			//BangHook = new LibPDBangHook(RaiseBangEvent);
+			//PInvoke.set_banghook(BangHook);
+
+			//FloatHook = new LibPDFloatHook(RaiseFloatEvent);
+			//PInvoke.set_floathook(FloatHook);
+
+			//SymbolHook = new LibPDSymbolHook(RaiseSymbolEvent);
+			//PInvoke.set_symbolhook(SymbolHook);
+
+			//ListHook = new LibPDListHook(RaiseListEvent);
+			//PInvoke.set_listhook(ListHook);
+
+			//MessageHook = new LibPDMessageHook(RaiseMessageEvent);
+			//PInvoke.set_messagehook(MessageHook);
+		}
+
+		~Pd ()
+		{
+			Dispose (false);
+		}
+
+		public void Dispose ()
+		{
+			Dispose (true);
+			GC.SuppressFinalize (this);
+		}
+
+		private void Dispose (bool disposing)
+		{
+			if (disposing) {
+				Stop ();
+			}
+		}
+
+		/// <summary>
+		/// Starts audio computation.
+		/// </summary>
+		public void Start ()
+		{
+			PInvoke.init_audio (Inputs, Outputs, SampleRate);
+			SendMessage ("pd", "dsp", 1);
+			IsComputing = true;
+		}
+
+		private void SendMessage (string receiver, string message, params object[] args)
+		{
+			SendArgs (args);
+			var finish = PInvoke.finish_message (receiver, message);
+
+			if (finish != 0) {
+				throw new PdProcessException (finish, "finish_message");
+			}
+		}
+
+		private static void SendArgs (object[] args)
+		{
+			int startMessage = PInvoke.start_message (args.Length);
+			if (startMessage != 0) {
+				throw new PdProcessException (startMessage, "start_message");
+			}
+			foreach (object arg in args) {
+				if (arg is int?) {
+					PInvoke.add_float ((int)((int?)arg));
+				} else if (arg is float?) {
+					PInvoke.add_float ((float)((float?)arg));
+				} else if (arg is double?) {
+					PInvoke.add_float ((float)((double?)arg));
+				} else if (arg is string) {
+					PInvoke.add_symbol ((string)arg);
+				} else {
+					throw new ArgumentOutOfRangeException ("args[]", arg, "Argument is of wrong type.");
+				}
+			}
+		}
+
+		public short[] Process (int ticks, short[] inBuffer)
+		{
+			short[] outBuffer = new short[Outputs * ticks * BlockSize];
+			PInvoke.process_short (ticks, inBuffer, outBuffer);
+			return outBuffer;
+		}
+
+		/// <summary>
+		/// Reads samples from inBuffer and processes them with Pd.
+		/// </summary>
+		/// <param name="ticks">Number of ticks to process. To reduce overhead of function calls, raise this number. To lower latency, reduce this number.</param>
+		/// <param name="inBuffer">Interleaved input buffer. Must be at least of size [InputChannels] * [Blocksize] * [ticks]</param>
+		/// <returns>Interleaved processed audio samples.</returns>
+		public float[] Process (int ticks, float[] inBuffer)
```

Comment:
- The `libpd_process_*` functions do never return anything different than 0 (see in C code).
- All pointer versions have been removed. Has anyone ever used them?
- Returning a new array looks more natural to me
- Should ticks or inBuffer be the first parameter?